staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go

```diff
@@ -126,3 +151,182 @@ func NewDefaultConfig(prefix string, codec runtime.Codec) *Config {
+
+	var (
+		client *clientv3.Client
+		err    error
+	)
+	if pollErr := wait.PollImmediateUntilWithContext(ctx, time.Second, func(ctx context.Context) (bool, error) {
+		client, err = getClient(ctx, c)
+		if err != nil {
+			klog.ErrorS(err, "failed to get etcd client")
+		}
+		return err == nil, nil
+	}); pollErr != nil {
+		return nil, fmt.Errorf("failed to get etcd client, pollErr=%v: %w", pollErr, err)
+	}
+
+	return client, err
+}
+
+func getClient(ctx context.Context, c TransportConfig) (*clientv3.Client, error) {
+	tlsInfo := transport.TLSInfo{
+		CertFile:      c.CertFile,
+		KeyFile:       c.KeyFile,
+		TrustedCAFile: c.TrustedCAFile,
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	// NOTE: Client relies on nil tlsConfig
+	// for non-secure connections, update the implicit variable
+	if len(c.CertFile) == 0 && len(c.KeyFile) == 0 && len(c.TrustedCAFile) == 0 {
+		tlsConfig = nil
+	}
+	networkContext := egressselector.Etcd.AsNetworkContext()
+	var egressDialer utilnet.DialFunc
+	if c.EgressLookup != nil {
+		egressDialer, err = c.EgressLookup(networkContext)
+		if err != nil {
+			return nil, err
+		}
+	}
+	dialOptions := []grpc.DialOption{
+		grpc.WithBlock(), // block until the underlying connection is up
+		// use chained interceptors so that the default (retry and backoff) interceptors are added.
+		// otherwise they will be overwritten by the metric interceptor.
+		//
+		// these optional interceptors will be placed after the default ones.
+		// which seems to be what we want as the metrics will be collected on each attempt (retry)
+		grpc.WithChainUnaryInterceptor(grpcprom.UnaryClientInterceptor),
+		grpc.WithChainStreamInterceptor(grpcprom.StreamClientInterceptor),
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIServerTracing) {
+		tracingOpts := []otelgrpc.Option{
+			otelgrpc.WithPropagators(tracing.Propagators()),
+			otelgrpc.WithTracerProvider(c.TracerProvider),
+		}
+		// Even with Noop  TracerProvider, the otelgrpc still handles context propagation.
+		// See https://github.com/open-telemetry/opentelemetry-go/tree/main/example/passthrough
+		dialOptions = append(dialOptions,
+			grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor(tracingOpts...)),
+			grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor(tracingOpts...)))
+	}
+	if egressDialer != nil {
+		dialer := func(ctx context.Context, addr string) (net.Conn, error) {
+			if strings.Contains(addr, "//") {
+				// etcd client prior to 3.5 passed URLs to dialer, normalize to address
+				u, err := url.Parse(addr)
+				if err != nil {
+					return nil, err
+				}
+				addr = u.Host
+			}
+			return egressDialer(ctx, "tcp", addr)
+		}
+		dialOptions = append(dialOptions, grpc.WithContextDialer(dialer))
+	}
+
+	l, err := logutil.CreateDefaultZapLogger(etcdClientDebugLevel())
+	if err != nil {
+		return nil, err
+	}
+	etcd3ClientLogger := l.Named("etcd-client")
```

Comment:
This was introduced in https://github.com/kubernetes/kubernetes/pull/111477, as a workaround for the excessive memory consumed by creating many etcd client loggers. Given that this PR causes Kubernetes to create only one per transport it should be safe to return to letting etcd use its default client (by not passing in `Logger` below).

Reply:
Ah nice, let me do that.